pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -459,9 +459,10 @@ func (a *Actuator) getNodeReference(scope *actuators.MachineScope) (*apicorev1.O
 			// TODO(vincepri): Improve this comparison without relying on substrings.
 			if strings.Contains(node.Spec.ProviderID, instanceID) {
 				return &apicorev1.ObjectReference{
-					Kind:       node.Kind,
-					APIVersion: node.APIVersion,
+					Kind:       "Node",
```

Comment:
Isn't node.Kind `Node`? What about the APIVersion change?

Reply:
I thought so as well, but if you look at the output of @ashish-amarnath's testing you will see that previously Kind and APIVersion were unset: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/214#issuecomment-473615702